lib/index.js

```diff
@@ -677,12 +724,50 @@ FSTree.prototype.symlinkSync = function(target, relativePath /*, type */) {
 
   // TODO: should we read the file here so our entry has size, mode & checksum?
   // this turns 1 io -> 2 io (symlink -> symlink + stat)
-  let entry = new Entry(normalizedPath, 0, Date.now(), 0);
+
+
+  let mode = Entry.SYMLINK_MODE;
+
+  let entry = new Entry(normalizedPath, 0, Date.now(), mode);
   let operation = result.entry ? 'change' : 'create';
+
+  entry._projection = srcEntry._projection;
+
   this._track(operation, entry);
   this._insertAt(result, entry);
 };
 
+FSTree.prototype.symlinkSyncFromInput = function(srcFSTree,  srcAbsolutePath,  destRelativePath, isLinkDir /*, type */) {
```

Comment:
what are the semantics of `isLinkDir`?  why is this a `symlinkSyncFromInput` concern and not a merge trees concern?